core/src/main/java/org/apache/iceberg/TableScanContext.java

```diff
@@ -75,7 +77,8 @@ private TableScanContext(
       Long toSnapshotId,
       ExecutorService planExecutor,
       boolean fromSnapshotInclusive,
-      ScanReporter scanReporter) {
+      ScanReporter scanReporter,
+      String ref) {
```

Comment:
I don't think it is necessary to add this, since the ref can really only select a snapshot. We can use the existing `snapshot` behavior.

Reply:
It was required before because if we do useRef("branch").asOfTime(someTime) the context acts as a placeholder to store the information for branches, so that in the asOfTime we can choose the right ref. If we don't do time travel with branches then yes we just need to set a snapshot.